src/PrivateKey.cs

```diff
@@ -39,6 +39,13 @@ public byte[] Sign(byte[] hash)
         {
             var hash32 = new uint256(hash);
             var signature = _bitcoinKey.SignCompact(hash32, false);
```

Comment:
The if statement should be operating on the private key first before SignCompact.

if(compressed)
> OperateOnPrivateKey
SignCompact